src/test/java/com/google/sps/TripTest.java

```diff
@@ -0,0 +1,125 @@
+  private static final String TRIP_NAME = "Trip to California";
+  private static final String START_DATE = "2020-02-29";
+  private static final String END_DATE = "2020-03-05";
+  private static final int NUM_DAYS = 6;
+  
+  private ArrayList<TripDay> tripDays;
+
+  private static final String TIMES_SQUARE_ID = "ChIJmQJIxlVYwokRLgeuocVOGVU";
+  private static final String CENTRAL_PARK_ID = "ChIJ4zGFAZpYwokRGUGph3Mf37k";
+  private static final String WORLD_TRADE_ID = "ChIJy7cGfBlawokR5l2e93hsoEA";
+  private static final String EMPIRE_STATE_ID = "ChIJtcaxrqlZwokRfwmmibzPsTU";
+  private static final String HOTEL_ID = "ChIJ68J3tfpYwokR2HaRoBcB4xg";
+
+  private ArrayList<String> locations = new ArrayList<>(Arrays.asList(TIMES_SQUARE_ID, CENTRAL_PARK_ID,
+                                                                    WORLD_TRADE_ID, EMPIRE_STATE_ID));
+
+  // Test the trip constructor which allows for multiple days
+  @Test
+  public void testTripConstructorMultiDay() {
+    TripDay tripDay1 = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    TripDay tripDay2 = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    
+    tripDays = new ArrayList<>();
+    tripDays.add(tripDay1);
+    tripDays.add(tripDay2);
+
+    Trip trip = new Trip(TRIP_NAME, START_DATE, END_DATE, tripDays);
+
+    Assert.assertEquals(trip.getTripName(), TRIP_NAME);
+    Assert.assertEquals(trip.getStartDate(), START_DATE);
+    Assert.assertEquals(trip.getEndDate(), END_DATE);
+    Assert.assertEquals(trip.getNumDays(), NUM_DAYS);
+    Assert.assertEquals(trip.getTripDays(), tripDays);
+  }
+
+  // Test the Trip constructor for a single day (MVP version)
+  @Test
+  public void testTripConstructorSingleDay() {
+    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    
+    tripDays = new ArrayList<>();
+    tripDays.add(tripDay);
+
+    Trip trip = new Trip(TRIP_NAME, START_DATE, tripDays);
+
+    Assert.assertEquals(trip.getTripName(), TRIP_NAME);
+    Assert.assertEquals(trip.getStartDate(), START_DATE);
+    Assert.assertEquals(trip.getEndDate(), START_DATE);
+    Assert.assertEquals(trip.getNumDays(), 1);
+    Assert.assertEquals(trip.getTripDays(), tripDays);
+    
+  }
+
+  // Test multiple day constructor with null parameter
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripConstructorMultiDayNull() {
+    Trip trip = new Trip(TRIP_NAME, START_DATE, END_DATE, null);
+  }
+
+  // Test multiple day constructor with wrong date format
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripConstructorMultiDayInvalidDate() {
+    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, locations);
+    
+    tripDays = new ArrayList<>();
+    tripDays.add(tripDay);
+
+    Trip trip = new Trip(TRIP_NAME, "06-28-2020", END_DATE, tripDays);
+  }
+
+  // Test multiple day constructor with over a month
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripConstructorMultiDayLongTrip() {
```

Comment:
this is great, since it captures the edge case of being at 31 days, but you also need a test for the edge case of it's '0'. That test would have caught your off-by-one bug in Trip. Whenever you have restrictions like that in your code, it's important to test right on the edge of those restrictions

Reply:
I still don't see a test that checks for trip of length 0? Am I missing it?

Reply:
Good catch! Because a trip with the same start and end date is considered to have 1 day, a trip of length 0 would occur when the end date is one day before the start date. I added a test for this to ensure that the IllegalArgumentException is thrown.